### PR TITLE
Make pp.Decoder.MaxLength configurable

### PR DESCRIPTION
**Pull request by AskAlexSharov**
For https://github.com/anacrolix/torrent/discussions/741

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -155,6 +156,10 @@ type ClientConfig struct {
 	DisableWebtorrent bool
 	DisableWebseeds   bool
 
+	// Drop any peer response bigger than this. To prevent DoS or abuse.
+	// TorrentSpec.ChunkSize can't be bigger than this value
+	MaxResponseLength pp.Integer
```
Comment (anacrolix) [inline config.go:161]: How about `DecodePeerMessageMaxLength`?

Comment (anacrolix) [reply config.go:161]: Or `PeerMessageDecoderMaxLength`

**Comment by AskAlexSharov** [comment]
renamed to PeerMessageDecoderMaxLength, fixed tests

**Comment by anacrolix** [comment]
Very nice! Just before I squash and merge it @AskAlexSharov : any reason not to just set the max decoder length based on the Torrent chunk size? (say max(4*chunkSize, ~24KiB) or so?

**Comment by AskAlexSharov** [comment]
@anacrolix ok, changed to `4 * max(t.chunkSize, defaultChunkSize)` and removed global config

**Comment by anacrolix** [comment]
Thanks, it's probably simpler not to expose the config option and infer it automatically like you've done now. I think the config is already complex enough. I'll merge this when the tests pass.

**Pull request merged**